File: javascript/json-transform/src/functions/TransformerFunctionAnd.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { isTruthy } from "../JsonHelpers";

class TransformerFunctionAnd extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const arr = await context.getJsonElementStreamer(null);
    if (arr == null) {
      return false;
    }
    return arr.stream().all(item => isTruthy(item));
  }
}

export default TransformerFunctionAnd;

File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionAnd.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionAnd extends TransformerFunction {
    public TransformerFunctionAnd() {
        super();
    }

    @Override
    public Object apply(FunctionContext context) {
        var adapter = context.getAdapter();
        var value = context.getJsonElementStreamer(null);
        if (value == null) {
            return false;
        }
        return value.stream().allMatch(adapter::isTruthy);
    }
}
